### DISCUSSION TITLE
Support for Secrets Transform Engine?

### DISCUSSION THREAD

**Discussion by gregoryenglish** [quarkiverse/quarkus-vault#294] [General]
Does the current version support encode/decode with the Vault Secrets Transform engine? https://developer.hashicorp.com/vault/api-docs/secret/transform

**Comment by kdubb** [comment]
The entire Vault Transform engine is supported by the new Vault Client (version 4+). The traditional Quarkus interfaces for Transform do not support everything, but I'm fairly certain they support those endpoints.

If you need to use the new Vault client directly you can `@Inject VaultClient vaultClient`. It's what the traditional Quarkus interfaces use under the hood in version 4+.
